src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -411,15 +411,22 @@ uct_ib_mlx5_post_send(uct_ib_mlx5_txwq_t *wq,
     /* We don't want the compiler to reorder instructions and hurt latency */
     ucs_compiler_fence();
 
-    /* Advance queue pointer */
+    /*
+     * Advance queue pointer.
+     * We return the number of BBs the *previous* WQE has consumed, since CQEs
+     * are reporting the index of the first BB rather than the last. We have
+     * reserved QP space for at least UCT_IB_MLX5_MAX_BB to accommodate.
+     * */
     ucs_assert(ctrl == wq->curr);
-    wq->curr       = src;
-    wq->prev_sw_pi = wq->sw_pi;
-    wq->sw_pi      = sw_pi;
+    res_count    = wq->sw_pi - wq->prev_sw_pi;
```

Comment:
Probably I missed smth but `wq->prev_sw_pi` is initialized with `-1` (UINT16_MAX) at start, is it valid for 1st posting?
pls align with below